examples/poem_example/api/src/lib.rs

```diff
@@ -30,24 +31,22 @@ struct Params {
 }
 
 #[handler]
-async fn create(state: Data<&AppState>, form: Form<post::Model>) -> Result<impl IntoResponse> {
+async fn create(txn: Data<&Arc<DatabaseTransaction>>, form: Form<post::Model>) -> Result<impl IntoResponse> {
     let form = form.0;
-    let conn = &state.conn;
 
-    MutationCore::create_post(conn, form)
+    MutationCore::create_post(txn.0, form)
```

Comment:
Hey @TheCataliasTNT2k, I think you can start a transaction with 
```rs
let conn = &state.conn;

let txn = conn.begin().await?;

MutationCore::create_post(&txn, form)
```

Reply:
Yes, this is possible, but the idea was, that you do not need to worry about committing or anything like that.
Use the transaction as you want, 
return OK for commit
return Error for rollback